sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/Distribute.java

```diff
@@ -452,10 +487,17 @@ public synchronized void onActivityCreated(Activity activity, Bundle savedInstan
         /* Clear workflow finished state if launch recreated, to achieve check on "startup". */
         if (activity.getClass().getName().equals(mLauncherActivityClassName)) {
             AppCenterLog.info(LOG_TAG, "Launcher activity restarted.");
-            if (mChannel != null && getStoredDownloadState() == DOWNLOAD_STATE_COMPLETED) {
-                mWorkflowCompleted = false;
-                mBrowserOpenedOrAborted = false;
-            }
+            resetWorkflow();
+        }
+    }
+
+    /**
+     * Reset current state to allow a new update check.
+     */
+    private void resetWorkflow() {
+        if (mChannel != null && getStoredDownloadState() == DOWNLOAD_STATE_COMPLETED) {
```

Comment:
This kind of check is what prevents the UI from switching to the new value @clpolet. Resume is checking for that too. I tested in app with a timer and got:

> 2020-01-21 16:40:35.053 9757-9757/com.microsoft.appcenter.sasquatch.project D/AppCenterDistribute: Previous dialog is still being shown in the same activity.
> 2020-01-21 16:40:35.053 9757-9757/com.microsoft.appcenter.sasquatch.project V/AppCenterDistribute: Already checking or checked latest release.

Looks like we are good.